inc/utilities.h

```diff
@@ -6,6 +6,9 @@
 #include <pthread.h>
 #include "stdint.h"
 #include "assert.h"
+#include "SDL2/SDL.h"
+
+#define THREAD_SLEEP(ms) SDL_Delay(ms)
```

Comment:
Why did I have to make it as a preprocessor macro? It is prune to bugs due to the lack of argument type check. I better change it into an Inline func

Reply:
Done, not as an inline due to C99 inline semantics requiring me to mark func as either extern or static, both of which i regarded as a code smell (especially static, due to static function usually meant for just one translation unit, whoich is not the case here)